front/pages/api/w/[wId]/data_sources/request_access.ts

```diff
@@ -0,0 +1,136 @@
+import type { Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import { isLeft } from "fp-ts/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import config from "@app/lib/api/config";
+import { withSessionAuthenticationForWorkspace } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { sendEmail } from "@app/lib/email";
+import logger from "@app/logger/logger";
+import { apiError } from "@app/logger/withlogging";
+
+export const PostRequestAccessBodySchema = t.type({
+  emailMessage: t.string,
+  email: t.string,
+  dataSourceName: t.string,
+});
+
+export type PostRequestAccessBody = t.TypeOf<
+  typeof PostRequestAccessBodySchema
+>;
+
+async function sendEmailWithTemplate(
+  to: string,
+  from: { name: string; email: string },
+  subject: string,
+  body: string
+): Promise<Result<void, Error>> {
```

Comment:
When we only deal with native types for params we should try to use object instead.